File: client/src/app/_models/bracketTemplate.ts
// client/src/app/_models/bracketTemplate.ts - ENHANCED VERSION

import { BracketType } from "./bracketType";
import { Team } from "./team";

export interface BracketTemplate {
  id: number;
  name: string;
  officialTemplate: boolean;
  numberOfRounds: number;
  numberOfBrackets: number;  // Add this property
  createdAt?: Date;
  updatedAt?: Date;
  bracketType: BracketType;
  teams: Team[];
}

// Helper interface for bracket template creation
export interface CreateBracketTemplateDTO {
  name: string;
  numberOfRounds: number;
  bracketType: BracketType;
}

// Helper interface for bracket template update
export interface UpdateBracketTemplateDTO {
  id: number;
  name: string;
  numberOfRounds: number;
  bracketType: BracketType;
}

File: API/Entities/BracketTemplate.cs
using System;

namespace API.Entities;

public class BracketTemplate
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool OfficialTemplate { get; set; } = false;
    public int NumberOfRounds { get; set; }
    public int NumberOfBrackets { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public BracketType BracketType { get; set; } = BracketType.SingleTeam;

    public int UserId { get; set; }
    public AppUser User { get; set; } = null!;

    public ICollection<PostBracket> PostBrackets { get; set; } = [];
    public ICollection<Team> Teams { get; set; } = [];

    public BracketTemplate(int numberOfRounds)
    {
        NumberOfRounds = numberOfRounds;
        NumberOfBrackets = (int)Math.Pow(2, numberOfRounds);
    }

    public void AddNewPostBracket()
    {
        PostBrackets.Add(new PostBracket(BracketType, NumberOfBrackets));
    }
}
